src/validation_workflow/docker/validation_docker.py

```
@@ -147,17 +195,41 @@ def run_container(self, OpenSearch_image: str, OpenSearchDashboard_image: str, v
+                shutil.copy2(source_file, self.target_yml_file)
 
-        self.inplace_change(self.target_yml_file, f'opensearchproject/opensearch:{version[0]}', f'{OpenSearch_image}:{version}')
-        self.inplace_change(self.target_yml_file, f'opensearchproject/opensearch-dashboards:{version[0]}', f'{OpenSearchDashboard_image}:{version}')
+            if (self.args.using_staging_artifact_only):
+                self.images_used = f'{OpenSearch_image}:{version}'
+            else:
+                self.images_used = f'{OpenSearch_image}:{version}'
+
+            self.inplace_change(
+                self.target_yml_file,
+                f'opensearchproject/opensearch:{version[0]}',
+                f'{OpenSearch_image}:{version}.{self.args.os_build_number}'
+                if self.args.using_staging_artifact_only
+                else f'{OpenSearch_image}:{version}'
+            )
 
-        # spin up containers
-        docker_compose_up = f'docker-compose -f {self.target_yml_file} up -d'
-        result = subprocess.run(docker_compose_up, shell=True, stdout=PIPE, stderr=PIPE, universal_newlines=True)
-        return ('returncode=0' in (str(result)), self.target_yml_file)
+            self.inplace_change(
+                self.target_yml_file,
+                f'opensearchproject/opensearch-dashboards:{version[0]}',
+                f'{OpenSearchDashboard_image}:{version}.{self.args.osd_build_number}'
+                if self.args.using_staging_artifact_only
+                else f'{OpenSearchDashboard_image}:{version}'
+            )
+            # spin up containers
+            self.docker_compose_up = f'docker-compose -f {self.target_yml_file} up -d'
+            result = subprocess.run(self.docker_compose_up, shell=True, stdout=PIPE, stderr=PIPE, universal_newlines=True)
+            return ('returncode=0' in (str(result)), self.target_yml_file)
+
+        except Exception as e:
+            raise Exception(f"An error occurred: {e}")
```

There's no need to catch and re-raise, just let it raise an exception.

---

thanks, sure, let me update it in my next commit..